hydrolib/tools/ext_old_to_new/converters.py

```diff
@@ -119,7 +133,52 @@ class BoundaryConditionConverter(BaseConverter):
     def __init__(self):
         super().__init__()
 
-    def convert(self, forcing: ExtOldForcing) -> Boundary:
+    @staticmethod
+    def parse_tim_model(tim_files: List[TimModel], forcing: ExtOldForcing) -> TimModel:
+        """Parse the boundary condition related time series from the tim files.
+
+        Args:
+            tim_files (List[TimModel]): List of TIM models.
+            forcing (ExtOldForcing):
+
+        Returns:
+            TimModel: A TimModel object containing the time series data from the TIM files.
+        """
+        time_files_exist = all([tim_file.exists() for tim_file in tim_files])
```

Comment:
Actually, it is allowed to have a polyline file where only SOME of the points have .tim file.
So, imagine a polyline with 20 points. You may have only 3 .tim files: foo_0000.tim, foo_0014.tim and foo_19.tim, and that would be valid input.

The converted .bc file would have to contain three Forcing blocks, with names "foo_0000" / 14 / 19.